File: codegen-tool/src/main/java/com/dqjq/codegen/Generator/DtoGenerator.java
package com.dqjq.codegen.Generator;

import com.dqjq.codegen.CodegenConfig;
import com.dqjq.codegen.CodegenFieldDef;
import lombok.Getter;

import java.io.IOException;
import java.nio.file.Path;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

/**
 * DTO 类生成器
 * <p>
 * 负责生成数据传输对象 (DTO)。
 * </p>
 */
@Getter
public class DtoGenerator extends BaseGenerator {

    private Path dtoPath;

    public DtoGenerator(CodegenConfig cfg) {
        super(cfg);
    }

    @Override
    public void generate() throws IOException {
        Path baseOut = getBaseOutputDir();
        String basePackagePath = cfg.packageBase.replace('.', '/');
        String modulePath = baseOut + "/" + basePackagePath + "/" + cfg.module;
        Path dtoDir = projectRoot.resolve(modulePath + "/dto");
        
        ensureDirectory(dtoDir);
        
        this.dtoPath = dtoDir.resolve(cfg.entityName + "Dto.java");

        String template = readTemplate("dto.tpl");
        if (template.isEmpty()) {
            throw new IOException("无法加载 DTO 模板: dto.tpl");
        }

        String dtoPkg = cfg.packageBase + "." + cfg.module + ".dto";
        
        Set<String> imports = new HashSet<>();
        imports.add("lombok.Data");
        imports.add("lombok.EqualsAndHashCode");
        imports.add("io.swagger.annotations.ApiModelProperty");
        
        for (CodegenFieldDef f : cfg.fields) {
            if ("LocalDateTime".equals(f.type)) imports.add("java.time.LocalDateTime");
            if ("BigDecimal".equals(f.type)) imports.add("java.math.BigDecimal");
        }
        
        String dtoExtends = "";
        if (cfg.dtoBaseClass != null && !cfg.dtoBaseClass.isEmpty()) {
            imports.add(cfg.dtoBaseClass);
            dtoExtends = " extends " + simpleClassName(cfg.dtoBaseClass);
        }

        String fieldsContent = renderFields(cfg.fields);

        String content = template
                .replace("{{package}}", dtoPkg)
                .replace("{{imports}}", renderImports(imports))
                .replace("{{entityName}}", cfg.entityName)
                .replace("{{fields}}", fieldsContent)
                .replace("{{dtoExtends}}", dtoExtends);

        writeFile(dtoPath, content);
    }

    private String renderFields(List<CodegenFieldDef> fields) {
        StringBuilder sb = new StringBuilder();
        for (CodegenFieldDef f : fields) {
            sb.append("    @io.swagger.annotations.ApiModelProperty(value = \"").append(f.label).append("\")\n");
            sb.append("    private ").append(f.type).append(" ").append(f.name).append(";\n\n");
        }
        return sb.toString();
    }

}

File: codegen/codegen-tool-py/codegen/generators/dto_generator.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
DTO生成器
生成Java DTO类
"""

from pathlib import Path
from codegen.generators.base_generator import BaseGenerator

class DtoGenerator(BaseGenerator):
    """DTO生成器"""
    
    def generate(self):
        """
        生成DTO类
        """
        # 准备上下文数据
        context = {
            'package': self.get_full_package('dto'),
            'entity_name': self.code_generator.entity_name,
            'fields': self.code_generator.fields,
            'dto_base_class': self.config_dict.get('dtoBaseClass', ''),
            'has_base_class': bool(self.config_dict.get('dtoBaseClass', ''))
        }
        
        # 渲染模板
        content = self.render_template('java/dto.j2', context)
        
        # 生成文件路径
        output_dir = self.code_generator.get_output_path('dto')
        self.output_path = output_dir / f"{self.code_generator.entity_name}Dto.java"
        
        # 写入文件
        self.write_file(self.output_path, content)
